### PR TITLE
5665 bring over ExtractAction changes from file-search branch

### PR DESCRIPTION
**Pull request by wschaeferB**

### CODE REVIEW

File: Core/src/org/sleuthkit/autopsy/directorytree/actionhelpers/ExtractActionHelper.java [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+
+        /**
+         * Calculate the number of work units for the progress bar.
+         *
+         * @param file File whose children will be reviewed to get the number of
+         *             work units.
+         *
+         * @return The number of work units.
+         */
+        /*
```
Comment (kellykelly3) [inline Core/src/org/sleuthkit/autopsy/directorytree/actionhelpers/ExtractActionHelper.java]: Do you need this commented out function?

File: Core/src/org/sleuthkit/autopsy/directorytree/actionhelpers/ExtractActionHelper.java [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+     * @param fileExtractionTasks List of file extraction tasks.
+     */
+    private void runExtractionTasks(ActionEvent event, List<FileExtractionTask> fileExtractionTasks) {
+
+        // verify all of the sources and destinations are OK
+        for (Iterator<FileExtractionTask> it = fileExtractionTasks.iterator(); it.hasNext();) {
+            FileExtractionTask task = it.next();
+
+            if (ContentUtils.isDotDirectory(task.source)) {
+                //JOptionPane.showMessageDialog((Component) e.getSource(), "Cannot extract virtual " + task.source.getName() + " directory.", "File is Virtual Directory", JOptionPane.WARNING_MESSAGE);
```
Comment (kellykelly3) [inline Core/src/org/sleuthkit/autopsy/directorytree/actionhelpers/ExtractActionHelper.java]: Remove commented out code.

File: Core/src/org/sleuthkit/autopsy/directorytree/actionhelpers/ExtractActionHelper.java (hunk 1 above)
Comment (wschaeferB) [reply Core/src/org/sleuthkit/autopsy/directorytree/actionhelpers/ExtractActionHelper.java]: I left it as it was due to the @@@ style todo at line 306 which says to add this back in.

File: Core/src/org/sleuthkit/autopsy/directorytree/actionhelpers/ExtractActionHelper.java (hunk 2 above)
Comment (wschaeferB) [reply Core/src/org/sleuthkit/autopsy/directorytree/actionhelpers/ExtractActionHelper.java]: good catch.

**Pull request merged**